apps/vmq_server/src/vmq_queue.erl

```diff
@@ -1194,6 +1198,7 @@ maybe_expire_msg(SId,
     case vmq_time:is_past(ExpiryTS) of
         true ->
             on_message_drop_hook(SId, D, expired),
+            vmq_util:sample_log(1, "Message dropped for subscriber_id: ~p due to ~p", [SId, expired]),
```

Comment:
IMO we shouldn't log message drop for expiring clients. This would flood the error logs. The right way would be to add `on_message_drop` hook.